### PR TITLE
DOC: Adds ma.correlate examples

### PR DESCRIPTION
**Pull request by bmwoodruff**
I adapted the AI generated examples.
I removed the default parameters.
First example shows how ma.correlate adds a mask to regular arrays. Second example shows basic usage with defaults.
Third example introduces mixed arrays and other options.

[skip actions] [skip azp] [skip cirrus]

I think these changes provide a higher chance of the examples being not too trivial.  Anytime AI uses the default parameters, check that and remove them. Most of the changes I made were just to showcase more options and mixing masked/non masked arrays.

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -8160,18 +8160,18 @@ def correlate(a, v, mode='valid', propagate_mask=True):
...
     >>> v = np.ma.array([0, 1, 0])
-    >>> np.ma.correlate(a, v, mode='valid', propagate_mask=True)
+    >>> np.ma.correlate(a, v)
     masked_array(data=[--],
                  mask=[ True],
            fill_value=999999,
                 dtype=int64)
 
-    Correlation with different modes:
+    Correlation with different modes and mixed elements:
```
Comment (bmwoodruff) [inline numpy/ma/core.py:8169]:
```suggestion
    Correlation with different modes and mixed array types:
```

Comment (bmwoodruff) [reply numpy/ma/core.py:8169]: Just noticed I missed that. It's not elements, but array types.

**Review by otieno-juma — APPROVED** [review]
Thank you for reviewing the examples

**Review by otieno-juma — APPROVED** [review]
Thank you for reviewing

**Comment by bmwoodruff** [comment]
Looks like you'll still need to add the comment above about changing elements to mixed array types. 

Once that's all done, it would be great to squash this into a single commit, rather than 4. Try doing that on a new branch that you create based on this one. that way the commit trail is only 1 long.

**Comment by otieno-juma** [comment]
I have a feeling I might end up messing things up. I have 2 more commits on my branch for two more functions. Can we briefly meet up tomorrow?

**Comment by bmwoodruff** [comment]
Yep.  Let's meet then. This is why I was thinking "Create a new branch first, based off this branch". That way it won't mess things up.  And if it does get messed up, it's OK.  This is a great place to try, experiment, break something, etc.  

As long as you're working off a branch and don't change `main`, then you can always delete the branch and start again.  And if you mess up main, and aren't sure how to fix it, you can always delete your repo and reclone. 

If you like the current branch, then create a new one based on the branch you like. Then experiment with the new branch. If things go south, delete the new branch.

**Comment by bmwoodruff** [comment]
I bet you can do this. I opened up [Groq](https://console.groq.com/playground?model=llama3-70b-8192), and entered this user message:
```
I'm currently working a branch in git, call it `feature`.  I want to create a new branch based on `feature`, and then rebase the new branch to main, squashing all commits. This should leave me with a new branch that has all the same changes as `feature`, but only one commit away from main. Help.
```
The output follows:

----
You can achieve this by following these steps:

1. Create a new branch based on `feature`:
```
git checkout feature
git branch new-branch
```
2. Checkout the new branch:
```
git checkout new-branch
```
3. Rebase the new branch onto `main`, squashing all commits:
```
git rebase -i --onto main feature
```
This will open an interactive rebase session. You'll see a list of commits, and you can squash them all into one commit by replacing `pick` with `squash` for each commit, except for the first one, which should remain as `pick`.

For example:
```
pick <commit-hash> <commit-msg>
squash <commit-hash> <commit-msg>
squash <commit-hash> <commit-msg>
...
```
Save and close the editor to proceed with the rebase.

4. Once the rebase is complete, you'll have a new branch `new-branch` that has all the same changes as `feature`, but with only one commit away from `main`.

Note: If you want to avoid the interactive rebase session and squash all commits automatically, you can use:
```
git rebase -i --onto main feature --squash
```
This will squash all commits into one, but be careful when using this approach, as you won't have a chance to review the commits before squashing them.

----

I think the steps above are correct. the nice part is that as long as you create a new-branch first, it won't matter if you mess something up.   While sharing this with you, I learned better how to squash as well.

File: numpy/ma/core.py (hunk 1 above)
Comment (bmwoodruff) [reply numpy/ma/core.py:8169]: Looks like you missed adding this in. Once that gets done, I'll review this again and it looks like we can send it to Numpy as a PR. Ping me on the review task when this gets added in. https://github.com/possee-org/genai-numpy/issues/54

**Pull request merged**